File: Python3/3137_minimum_number_of_operations_to_make_word_k_periodic.py
# https://leetcode.cn/problems/minimum-number-of-operations-to-make-word-k-periodic/

class Solution:
    def minimumOperationsToMakeKPeriodic(self, word: str, k: int) -> int:
        n = len(word)
        cnt = defaultdict(int)
        cnt_max = 0
        for i in range(0, n, k):
            cnt[word[i: i+k]] += 1
            cnt_max = max(cnt_max, cnt[word[i: i+k]])
        return n // k - cnt_max

File: Cpp/3137_minimum_number_of_operations_to_make_word_k_periodic.cpp
// https://leetcode.cn/problems/minimum-number-of-operations-to-make-word-k-periodic/

class Solution {
public:
    int minimumOperationsToMakeKPeriodic(string word, int k) {
        unordered_map<string, int> cnt;
        int n = word.size(), cnt_max = 0;
        for (int i = 0; i < n; i += k) {
            string sub = word.substr(i, k);
            cnt[sub] += 1;
            cnt_max = max(cnt_max, cnt[sub]);
        }
        return n / k - cnt_max;
    }
};
